sn_networking/src/record_store.rs

```diff
@@ -56,6 +61,9 @@ pub struct NodeRecordStore {
     record_count_metric: Option<Gauge>,
     /// Counting how many times got paid
     received_payment_count: usize,
+    /// Encyption cipher for the records, randomly generated at node startup
+    /// Plus a 4 byte nonce starter
+    encryption_details: (Aes256GcmSiv, [u8; 4]),
```

Comment:
As I understand, this `encryption_details` shall be written to disk in case `restart of node` is allowed.
Otherwise, if a node restart with the same key, it won't be able to read the previously stored records.

ideally, I think the Aes cipher and the nonce shall be deduced from the private key instead, to avoid have to store extra stuff to disk.

Reply:
I was not imagining to store this to disk. Node restart here will necesitate wiping the data (I thought any restart is always with new keys at the moment, eg).